src/hidimstat/dcrt.py

```diff
@@ -7,304 +7,396 @@
+                        self.params_lasso_distillation_x
+                        if self.params_lasso_distillation_x is not None
+                        else self.params_lasso_screening
+                    ),
+                    params_lasso_distillation_y=(
+                        self.params_lasso_distillation_y
+                        if self.params_lasso_distillation_y is not None
+                        else self.params_lasso_screening
+                    ),
+                    fit_y=self.fit_y,
+                    n_jobs=1,  # the function is already called in parallel
+                    random_state=self.random_state,
+                )
+                for idx in selection_set
+            )
+        elif self.statistic == "random_forest":
+            selection_set = range(n_features)
+            self.non_selection = []
+            # For distillation of X use least_square loss
+            results = Parallel(self.n_jobs, verbose=self.joblib_verbose)(
+                delayed(_rf_distillation)(
+                    X_,
+                    y_,
+                    idx,
+                    sigma_X=self.sigma_X,
+                    n_tree=self.n_tree,
+                    problem_type=self.problem_type,
+                    n_jobs=1,  # the function is already called in parallel
+                    random_state=self.random_state,
+                    params_lasso_distillation_x=(
+                        self.params_lasso_distillation_x
+                        if self.params_lasso_distillation_y is not None
+                        else self.params_lasso_screening
+                    ),
+                )
+                for idx in selection_set
+            )
+        else:
+            raise ValueError(f"{self.statistic} statistic is not supported.")
+        # contatenate result
+        self.selection_features = np.ones((n_features,), dtype=bool)
+        self.selection_features[self.non_selection] = 0
+        self.X_residual = np.array([result[0] for result in results])
+        self.sigma2 = np.array([result[1] for result in results])
+        self.y_residual = np.array([result[2] for result in results])
+        self.clf_x_residual = np.array([result[3] for result in results])
+        self.clf_y_residual = np.array([result[4] for result in results])
+        return self
+
+    def _check_fit(self):
+        """
+        Check if the model has been fit before performing analysis.
+
+        This private method verifies that all necessary attributes have been set
+        during the fitting process.
+        These attributes include:
+        - selection_features
+        - X_residual
+        - sigma2
+        - y_residual
+        - clf_x_residual
+        - clf_y_residual
+
+        Raises
+        ------
+        ValueError
+            If any of the required attributes are missing, indicating the model
+            hasn't been fit.
+        """
+        if not (
```

Comment:
That's just a detail, but the class needs all these arguments rather than any of them, or am I mistaken?

Reply:
'NOT X AND NOT Y' is more readable ;-)